presto-hive/src/main/java/io/prestosql/plugin/hive/HiveBucketing.java

```diff
@@ -141,7 +143,8 @@ private static int hash(TypeInfo type, Block block, int position)
                         long bigintValue = prestoType.getLong(block, position);
                         return (int) ((bigintValue >>> 32) ^ bigintValue);
                     case FLOAT:
-                        return (int) prestoType.getLong(block, position);
+                        // convert to canonical NaN if necessary
```

Comment:
can you a test that for two different ints that represents NaN in float they are producing same hash? That would be much better than a comment in the code.

Reply:
I'd need a new connector where I can force a particular bits representation of a NaN value...

Reply:
Thanks for catching me. I added a unit test and it told me there's another method to update.